### PR TITLE
Prettydiff100

### PR DESCRIPTION
**Pull request by prettydiff**
This pull request inserts the latest version of Pretty Diff into Unibeautify.

I keep all the existing conventions in place as much as possible.  There is one override I added for safety.  If an option value is passed in that is the incorrect data type then I drop that option value and instead use the Pretty Diff default value.  This check is dynamic in a small function named `fixType`.

~~It looks like all the tests are passing now except for a 1% decrease in code coverage.  I will need some help with that one, but otherwise its ready to merge.~~

100% compliant and ready for merge pending your review.

### CODE REVIEW

File: test/index.spec.ts [hunk 1]
```diff
@@ -1,12 +1,57 @@
...
+      options: {
+        JavaScript: {
+          indent_char: " ",
+          indent_size: 2,
+        },
+      },
+      text,
+    })
+    .then(() => {
+      expect(typeof application.api.language.auto).toBe("function");
```
Comment (Glavin001) [inline test/index.spec.ts]: @prettydiff No need to export `application` as an alias for `prettydiff`. Just import `prettydiff` in the tests.

Comment (prettydiff) [reply test/index.spec.ts]: Please review again.

**Comment by Glavin001** [comment]
I'll take a closer look this weekend.

File: package.json [hunk 2]
```diff
@@ -32,15 +32,15 @@
   "homepage": "https://github.com/Unibeautify/beautifier-prettydiff#readme",
   "peerDependencies": {
     "unibeautify": ">= 0.11.2",
-    "prettydiff2": "^2.2.7"
+    "prettydiff": "^100.2.6"
```
[the remark this answers is not in the document]
Comment (prettydiff) [reply package.json]: Just let me know how I can help.  I work all day Saturday, but I have Friday mornings and all day Sunday free for writing code.

File: src/index.ts [hunk 3]
```diff
@@ -8,15 +8,30 @@ import {
 import * as readPkgUp from "read-pkg-up";
 import options from "./options";
 const { pkg } = readPkgUp.sync({ cwd: __dirname });
```
Comment (stevenzeck) [inline src/index.ts]: Change this line to the below
```suggestion
const pkg = readPkgUp.sync({ cwd: __dirname })!.package;
```

File: package.json (hunk 2 above)
[the remark this answers is not in the document]
Comment (stevenzeck) [reply package.json]: @Glavin001 why is this a breaking change?  As long as the `@beautifier/prettydiff` package version is majored when this is published, it shouldn't break anything.

File: src/index.ts (hunk 3 above)
Comment (prettydiff) [reply src/index.ts]: What is `!.package`?  My editor is telling me the reference `package` is not defined in the current file.  What conflicts in package-lock.json do you see?

Comment (prettydiff) [reply src/index.ts]: Disregard that previous comment.  I needed to update and fix some things.  All should be fine now.

Comment (stevenzeck) [reply src/index.ts]: Yeah sorry, the update to read-pkg-up and removing @types/read-pkg-up prompted this change.

**Comment by stevenzeck** [comment]
@Glavin001 can you take another look when you have a chance?  Once approved, merged, and published, we can then update the VSCode plugin.

**Pull request merged**